model/task/task.go

```diff
@@ -444,12 +444,14 @@ func (t *Task) DependenciesMet(depCaches map[string]Task) (bool, error) {
 		return true, nil
 	}
 
+	if !t.DependenciesMetTime.IsZero() && t.DependenciesMetTime != utility.ZeroTime {
```

Comment:
Ah, we actually have a function for this! https://github.com/evergreen-ci/evergreen/blob/28c66fda6b8f4013ed4e0b3548444b5cf0d09245/vendor/github.com/evergreen-ci/utility/time.go#L16